python/triton/language/extra/cuda/libdevice.py

```diff
@@ -1050,7 +1050,7 @@ def isnan(arg0, _builder=None):
         ], {
             (core.dtype("fp32"), ): ("__nv_isnanf", core.dtype("int32")),
             (core.dtype("fp64"), ): ("__nv_isnand", core.dtype("int32")),
-        }, is_pure=True, _builder=_builder)
+        }, is_pure=True, _builder=_builder).to(core.int1, _builder=_builder)
```

Comment:
I think this generates an extra instruction, is there a way to cast to int1 as a no-op? Or maybe it doesn't really matter?

Reply:
I think it doesn't really matter. When you use this, the compiler should be able to see that extra instruction and optimise it away if it can.

Reply:
> the compiler should be able to see that extra instruction and optimise it away if it can.

To reason about this it needs to know that the resulting value can only be 0 or 1. I wouldn't presume that's the case